pdp/doc_helpers/__init__.py

```diff
@@ -0,0 +1,172 @@
+"""
+This module provides helper functions for the user documentation build.
+This module is imported in setup.py to support Sphinx documentation builds
+using the custom setup.py command ``build_sphinx``.
+
+The helper functions support fetching additional documentation files from
+other project repos (namely, frontends that are not part of this repo)
+via HTTP requests to GitHub. A typical example is the Met Data Portal.
+
+The documentation is built with Sphinx. This module is not Sphinx-specific,
+although what it does is made necessary by how Sphinx works.
+
+The big picture is that each contributing external repo contains a manifest
+of files to download, and the files themselves. These files are downloaded
+from GitHub and saved to the Sphinx document source directory.
+
+Manifest and content files must be in a single directory, usually one
+dedicated to documentation.
+
+The manifest is a YAML file that lists files hierarchically, mirroring the
+structure of the source documentation directory. Files within a directory
+are represented by a list of file names; a (sub)directory is represented
+by a list item that is a dictionary whose key is the directory name and
+whose value is likewise a list of files or directories, recursively.
+
+For more details, see the README on this subject.
+"""
+from __future__ import print_function
+import subprocess
+from warnings import warn
+import urllib2
```

Comment:
You  may want to alert @eyvorchuk about the addition of this as it has implications for the 2-to-3 conversion that he'll need to work around.